fix crash: excepted singleton

lepistone opened the pull request.

lepistone committed:
fix crash: excepted singleton

pedrobaeza commented:
:+1:

yvaucher commented on stock_reserve/model/product.py:
```diff
@@ -60,7 +60,7 @@ class ProductProduct(models.Model):
+    @api.one
     def _reservation_count(self):
         StockReservation = self.env['stock.reservation']
         product_id = self._ids[0]
```
Could you replace it by self.id ? or use self.id directly in domain ?

yvaucher commented:
~~Tests are failing~~ just saw #42 

Can you improve the annoted line ?

lepistone commented:
I tried to add a little innocent test, and :snake: :snake: came out. I'm fixing this. Don't merge, please.

lepistone committed:
fix reservation_count computation

1. Make it float, because quantity is float.
2. Respect unit of measure (instead of adding apples and pears).
3. Do that for both product and template.
4. Test the above.

lepistone committed:
refactor: delegate to variants, use self.id

lepistone committed:
refactor tests, improve failures

lepistone commented:
We should be good now!

jgrandguillaume commented:
:+1:

yvaucher commented:
:+1:
